File: server/src/main/java/org/tbbtalent/server/service/CertificationService.java
package org.tbbtalent.server.service;

import org.tbbtalent.server.model.CandidateCertification;
import org.tbbtalent.server.request.certification.CreateCertificationRequest;

public interface CertificationService {

    CandidateCertification createCertification(CreateCertificationRequest request);

    void deleteCertification(Long id);

}

File: ui/candidate-portal/src/app/services/certification.service.ts
import {Injectable} from '@angular/core';
import {environment} from "../../environments/environment";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {CandidateCertification} from "../model/candidate-certification";

@Injectable({
  providedIn: 'root'
})
export class CertificationService {

  private apiUrl: string = environment.apiUrl + '/certification';

  constructor(private http: HttpClient) { }

  createCertification(request): Observable<CandidateCertification> {
    return this.http.post<CandidateCertification>(`${this.apiUrl}`, request);
  }

  deleteCertification(id: number): Observable<any> {
    return this.http.delete<any>(`${this.apiUrl}/${id}`);
  }
}
